config/api.yml

```diff
@@ -2336,7 +2336,17 @@
         - ems_infra_show_list
         - ems_block_storage_show_list
       - :name: create
-        :identifier: ems_infra_new
+        :identifiers:
+        - :klass: EmsInfra
+          :identifier: ems_infra_new
+        - :klass: EmsCloud
+          :identifier: ems_cloud_new
+        - :klass: Provider #EmsContainer
+          :identifier: ems_container_new
+        - :klass: EmsNetwork
+          :identifier: ems_network_new
+        - :klass: Provider
```

Comment:
`Provider` is in the list twice and with the `detect` above, I dont think you'd ever get to this one. :disappointed:

Reply:
I'm aware of that. But like I said, this isn't getting merged anyway, because it relies on the aliases and you nixed that.

Reply:
:+1: Nice, thanks!

Reply:
I still don't like this just being Provider

Reply:
Yeah, this one is special. https://www.manageiq.org/docs/reference/latest/api/reference/providers#foreman-support

Soon enough you'll be able to add providers of other types.  Maybe we should create a new generic "provider_add" under the API section of the product features YAML?  I'm okay with the way it is for this BZ.  @abellotti Thoughts?

Reply:
Yea, that makes sense.  Was thinking of the following for the API section, which will match closely to the api.yml but in MiqProductFeature lingo:

```
# Providers 
- :name: Providers
  :description: Everything under Providers
  :feature_type: node 
  :children:
  - :name: Create a new Provider
    :description: Create a new Provider
    :feature_type: admin
    :children:
    - :name: Create a new ManageIQ::Providers::InfraManager
      :description: Create a new ManageIQ Infrastructure Manager Provider
      :feature_type: admin
      :identifier: ems_infra_new
    - :name: Create a new ManageIQ::Providers::CloudManager
      :description: Create a new ManageIQ Cloud Manager Provider
      :feature_type: admin
      :identifier: ems_cloud_new
 ... 
```

We can iterate/refine above as needed.